### PR TITLE
Fix #136

### LINKED ISSUE #136: falcon-node-sensor after install logs show panic error (non olm install)
» kubectl -n falcon-operator logs -f deploy/falcon-operator-controller-manager -c manager
I1214 20:07:52.197322       1 request.go:655] Throttling request took 1.005292364s, request: GET:https://100.64.0.1:443/apis/monitoring.coreos.com/v1alpha1?timeout=32s
2021-12-14T20:07:52.403Z	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": ":8080"}
2021-12-14T20:07:52.403Z	INFO	setup	starting manager
I1214 20:07:52.403618       1 leaderelection.go:243] attempting to acquire leader lease falcon-operator/70435a7a.crowdstrike.com...
2021-12-14T20:07:52.403Z	INFO	controller-runtime.manager	starting metrics server	{"path": "/metrics"}
I1214 20:08:08.686119       1 leaderelection.go:253] successfully acquired lease falcon-operator/70435a7a.crowdstrike.com
2021-12-14T20:08:08.686Z	INFO	controller-runtime.manager.controller.falconnodesensor	Starting EventSource	{"reconciler group": "falcon.crowdstrike.com", "reconciler kind": "FalconNodeSensor", "source": "kind source: /, Kind="}
2021-12-14T20:08:08.686Z	INFO	controller-runtime.manager.controller.falconcontainer	Starting EventSource	{"reconciler group": "falcon.crowdstrike.com", "reconciler kind": "FalconContainer", "source": "kind source: /, Kind="}
2021-12-14T20:08:08.686Z	DEBUG	controller-runtime.manager.events	Normal	{"object": {"kind":"Lease","namespace":"falcon-operator","name":"70435a7a.crowdstrike.com","uid":"3a6cca45-43ad-4f0d-90a9-3f28c5d6644b","apiVersion":"coordination.k8s.io/v1","resourceVersion":"540620776"}, "reason": "LeaderElection", "message": "falcon-operator-controller-manager-65d47fdfbb-mcqk7_0f977c63-0210-47c3-8a45-b455d872fa7b became leader"}
2021-12-14T20:08:08.686Z	DEBUG	controller-runtime.manager.events	Normal	{"object": {"kind":"ConfigMap","namespace":"falcon-operator","name":"70435a7a.crowdstrike.com","uid":"faac05d0-b82b-471c-9f6c-b19fdd5dec04","apiVersion":"v1","resourceVersion":"540620775"}, "reason": "LeaderElection", "message": "falcon-operator-controller-manager-65d47fdfbb-mcqk7_0f977c63-0210-47c3-8a45-b455d872fa7b became leader"}
2021-12-14T20:08:08.786Z	INFO	controller-runtime.manager.controller.falconnodesensor	Starting EventSource	{"reconciler group": "falcon.crowdstrike.com", "reconciler kind": "FalconNodeSensor", "source": "kind source: /, Kind="}
2021-12-14T20:08:08.787Z	INFO	controller-runtime.manager.controller.falconcontainer	Starting Controller	{"reconciler group": "falcon.crowdstrike.com", "reconciler kind": "FalconContainer"}
2021-12-14T20:08:09.091Z	INFO	controller-runtime.manager.controller.falconcontainer	Starting workers	{"reconciler group": "falcon.crowdstrike.com", "reconciler kind": "FalconContainer", "worker count": 1}
2021-12-14T20:08:09.092Z	INFO	controller-runtime.manager.controller.falconnodesensor	Starting EventSource	{"reconciler group": "falcon.crowdstrike.com", "reconciler kind": "FalconNodeSensor", "source": "kind source: /, Kind="}
2021-12-14T20:08:09.192Z	INFO	controller-runtime.manager.controller.falconnodesensor	Starting Controller	{"reconciler group": "falcon.crowdstrike.com", "reconciler kind": "FalconNodeSensor"}
2021-12-14T20:08:09.192Z	INFO	controller-runtime.manager.controller.falconnodesensor	Starting workers	{"reconciler group": "falcon.crowdstrike.com", "reconciler kind": "FalconNodeSensor", "worker count": 1}
2021-12-14T20:08:09.192Z	INFO	controller-runtime.manager.controller.falconnodesensor	reconciling FalconNodeSensor	{"reconciler group": "falcon.crowdstrike.com", "reconciler kind": "FalconNodeSensor", "name": "falcon-node-sensor", "namespace": "falcon-node-sensor", "DaemonSet": "falcon-node-sensor/falcon-node-sensor"}
E1214 20:08:09.193179       1 runtime.go:78] Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(0x262d2e0), concrete:(*runtime._type)(0x253d3c0), asserted:(*runtime._type)(0x259b480), missingMethod:""} (interface conversion: interface {} is []interface {}, not string)
goroutine 347 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x26b4500, 0xc000a42630)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/runtime/runtime.go:48 +0x89
panic(0x26b4500, 0xc000a42630)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/crowdstrike/falcon-operator/pkg/common.FalconSensorConfig(0xc00004c158, 0xc000a424e0)
	/workspace/pkg/common/funcs.go:48 +0x368
github.com/crowdstrike/falcon-operator/pkg/assets/node.daemonsetConfigMap(0xc0008a8120, 0x19, 0xc0009bb480, 0x12, 0xc00004c158, 0x2a81e9d)
	/workspace/pkg/assets/node/configmap.go:29 +0x27c
github.com/crowdstrike/falcon-operator/pkg/assets/node.DaemonsetConfigMap(...)
	/workspace/pkg/assets/node/configmap.go:11
github.com/crowdstrike/falcon-operator/controllers/falcon.(*FalconNodeSensorReconciler).nodeSensorConfigmap(0xc0005e1470, 0xc0008a8120, 0x19, 0xc00004c000, 0x2e298a0, 0xc000809110, 0x19)
	/workspace/controllers/falcon/falconnodesensor_controller.go:171 +0x5c
github.com/crowdstrike/falcon-operator/controllers/falcon.(*FalconNodeSensorReconciler).handleConfigMaps(0xc0005e1470, 0x2e1b7e0, 0xc000a42390, 0xc00004c000, 0x2e298a0, 0xc000809110, 0x12, 0x2e507a0, 0xc00004c000, 0x0)
	/workspace/controllers/falcon/falconnodesensor_controller.go:147 +0x4c5
github.com/crowdstrike/falcon-operator/controllers/falcon.(*FalconNodeSensorReconciler).Reconcile(0xc0005e1470, 0x2e1b7e0, 0xc000a42390, 0xc0009bb480, 0x12, 0xc0009bb460, 0x12, 0xc000a42390, 0x40c11f, 0xc00004e000, ...)
	/workspace/controllers/falcon/falconnodesensor_controller.go:73 +0x368
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0007a4640, 0x2e1b720, 0xc000912d00, 0x27836e0, 0xc000a7c0c0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298 +0x317
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0007a4640, 0x2e1b720, 0xc000912d00, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x2e1b720, 0xc000912d00)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216 +0x4a
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000789f50)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0008d5f50, 0x2dc8300, 0xc000a422d0, 0xc000912d01, 0xc000736000)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000789f50, 0x3b9aca00, 0x0, 0x1, 0xc000736000)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x2e1b720, 0xc000912d00, 0xc0008090a0, 0x3b9aca00, 0x0, 0x1)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x2e1b720, 0xc000912d00, 0xc0008090a0, 0x3b9aca00)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:99 +0x57
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:213 +0x45e
panic: interface conversion: interface {} is []interface {}, not string [recovered]
	panic: interface conversion: interface {} is []interface {}, not string

goroutine 347 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x26b4500, 0xc000a42630)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/crowdstrike/falcon-operator/pkg/common.FalconSensorConfig(0xc00004c158, 0xc000a424e0)
	/workspace/pkg/common/funcs.go:48 +0x368
github.com/crowdstrike/falcon-operator/pkg/assets/node.daemonsetConfigMap(0xc0008a8120, 0x19, 0xc0009bb480, 0x12, 0xc00004c158, 0x2a81e9d)
	/workspace/pkg/assets/node/configmap.go:29 +0x27c
github.com/crowdstrike/falcon-operator/pkg/assets/node.DaemonsetConfigMap(...)
	/workspace/pkg/assets/node/configmap.go:11
github.com/crowdstrike/falcon-operator/controllers/falcon.(*FalconNodeSensorReconciler).nodeSensorConfigmap(0xc0005e1470, 0xc0008a8120, 0x19, 0xc00004c000, 0x2e298a0, 0xc000809110, 0x19)
	/workspace/controllers/falcon/falconnodesensor_controller.go:171 +0x5c
github.com/crowdstrike/falcon-operator/controllers/falcon.(*FalconNodeSensorReconciler).handleConfigMaps(0xc0005e1470, 0x2e1b7e0, 0xc000a42390, 0xc00004c000, 0x2e298a0, 0xc000809110, 0x12, 0x2e507a0, 0xc00004c000, 0x0)
	/workspace/controllers/falcon/falconnodesensor_controller.go:147 +0x4c5
github.com/crowdstrike/falcon-operator/controllers/falcon.(*FalconNodeSensorReconciler).Reconcile(0xc0005e1470, 0x2e1b7e0, 0xc000a42390, 0xc0009bb480, 0x12, 0xc0009bb460, 0x12, 0xc000a42390, 0x40c11f, 0xc00004e000, ...)
	/workspace/controllers/falcon/falconnodesensor_controller.go:73 +0x368
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0007a4640, 0x2e1b720, 0xc000912d00, 0x27836e0, 0xc000a7c0c0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298 +0x317
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0007a4640, 0x2e1b720, 0xc000912d00, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x2e1b720, 0xc000912d00)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216 +0x4a
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000789f50)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0008d5f50, 0x2dc8300, 0xc000a422d0, 0xc000912d01, 0xc000736000)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000789f50, 0x3b9aca00, 0x0, 0x1, 0xc000736000)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x2e1b720, 0xc000912d00, 0xc0008090a0, 0x3b9aca00, 0x0, 0x1)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x2e1b720, 0xc000912d00, 0xc0008090a0, 0x3b9aca00)
	/go/pkg/mod/k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:99 +0x57
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:213 +0x45e

### PR DESCRIPTION
**Pull request by isimluk**

### CODE REVIEW

**Comment by isimluk** [comment]
@redhatrises, This fixes the crash in #136.

However, we still need fix to properly support tags. Would You mind If I rewrote dynamic conversion (https://github.com/CrowdStrike/falcon-operator/blob/ad1ee5361b398398e15bd880d2a3f6874f40c7c0/pkg/common/funcs.go#L28-L36 ) with explicit conversion?

**Comment by isimluk** [comment]
After this PR, operator won't crash but reports:

```
2021-12-16T12:38:45.503Z        ERROR   controller-runtime.manager.controller.falconnodesensor  Reconciler error        {"reconciler group": "falcon.crowdstrike.com", "reconciler kind": "FalconNodeSensor", "name": "falcon-node-sensor", "namespace": "default",
"error": "unexpected type received for FALCONCTL_OPT, field 'tags', type: []interface {}, value: [test]"}
```

File: controllers/falcon/falconnodesensor_controller.go [hunk 1]
```diff
@@ -155,7 +160,12 @@ func (r *FalconNodeSensorReconciler) handleConfigMaps(ctx context.Context, nodes
 		logger.Error(err, "error getting Configmap")
 		return nil, updated, err
 	} else {
-		err = r.Update(ctx, r.nodeSensorConfigmap(cmName, nodesensor, logger))
+		configmap, err := r.nodeSensorConfigmap(cmName, nodesensor)
+		if err != nil {
+			logger.Error(err, "Failed to format Configmap for update", "Configmap.Namespace", nodesensor.Namespace, "Configmap.Name", cmName)
+			return nil, updated, err
+		}
+		err = r.Update(ctx, configmap)
```
Comment (redhatrises) [inline controllers/falcon/falconnodesensor_controller.go:168]: Why not move the error message into `r.nodeSensorConfigmap`?

Comment (isimluk) [reply controllers/falcon/falconnodesensor_controller.go:168]: Better to bring it back to the controller. The higher we bring it the better. One day we may need to create events or report the error into the `status:` subresource.

File: pkg/assets/node/configmap.go [hunk 2]
```diff
@@ -7,11 +7,11 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func DaemonsetConfigMap(dsname string, nsname string, falconsensor *falconv1alpha1.FalconSensor) *corev1.ConfigMap {
-	return daemonsetConfigMap(dsname, nsname, falconsensor)
-}
-
-func daemonsetConfigMap(dsname string, nsname string, falconsensor *falconv1alpha1.FalconSensor) *corev1.ConfigMap {
```
Comment (redhatrises) [inline pkg/assets/node/configmap.go:14]: Let's keep this function as it allows us to minimize the amount of functions in the future if we have different types of configmaps here.

Comment (isimluk) [reply pkg/assets/node/configmap.go:14]: Let's keep an extra function today, so we can minimise amount of functions in future. Please explain.

**Pull request merged**